src/dos/drive_local.cpp

```diff
@@ -46,32 +47,53 @@
 #include "cross.h"
 #include "inout.h"
 
-bool localDrive::FileCreate(DOS_File * * file,char * name,uint16_t /*attributes*/) {
-//TODO Maybe care for attributes but not likely
+bool localDrive::FileCreate(DOS_File * * file,char * name,uint16_t attributes) {
+	// Don't allow overwriting read-only files.
+	uint16_t test_attr = 0;
+	if (GetFileAttr(name, &test_attr) && (test_attr & DOS_ATTR_READ_ONLY)) {
```

Comment:
Going forward, we should use the nice `FatAttributeFlags` kcgen introduced. @weirddan455 has made good use of it in his MOVE command implementation, you can refer to that for example usages.